libs/sdk-flutter/android/src/main/kotlin/com/breez/breez_sdk/BreezSDKPlugin.kt

```diff
@@ -1,35 +1,97 @@
 import io.flutter.plugin.common.MethodChannel.Result
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 
 /** BreezSDKPlugin */
-class BreezSDKPlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "breez_sdk")
-    channel.setMethodCallHandler(this)
-  }
-
-  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+class BreezSDKPlugin : FlutterPlugin, MethodCallHandler, EventChannel.StreamHandler {
+    /// The MethodChannel that will the communication between Flutter and native Android
+    ///
+    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+    /// when the Flutter Engine is detached from the Activity
+    private lateinit var channel: MethodChannel
+    private var eventChannel: EventChannel? = null
+    private var eventSink: EventChannel.EventSink? = null
+    private var nodeLogStream: LogStream? = null
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "breez_sdk")
+        channel.setMethodCallHandler(this)
+
+        eventChannel =
+            EventChannel(flutterPluginBinding.binaryMessenger, "breez_sdk_node_logs")
+        setNodeLogStream()
+        eventChannel?.setStreamHandler(this)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        if (call.method == "getPlatformVersion") {
+            result.success("Android ${android.os.Build.VERSION.RELEASE}")
+        } else {
+            result.notImplemented()
+        }
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
     }
-  }
 
-  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
-    channel.setMethodCallHandler(null)
-  }
+
+    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+        eventSink = events
+    }
+
+    override fun onCancel(arguments: Any?) {
+        eventSink = null
+        eventChannel = null
+    }
+
+    private fun setNodeLogStream(): LogStream? {
```

Comment:
As is I don't see a way we can get the log stream from the worker unless we have a public method somewhere that can be called both from the worker and the plugin.

Reply:
`LogEntry`'s are emitted via a `SharedFlow` and we `subscribe()` to it to listen to it from the worker.

I didn't had a chance to test these changes but you're right, the node log stream being used should be public and we should access that instance's subscribe method.

Reply:
So perhaps taking the SdkListener outside the plugin and Adding the ability to static initialize it (and subscribe to the sdk) via a public static method that can be used from both the flutter plugin and the worker.
The MutableShareflow seems a right decision and every side may add its own subscription to it?

Reply:
That's right, there can be many subscribers to the log event flow.